.github/workflows/test.yml

```diff
@@ -52,37 +52,38 @@ jobs:
           mkdir -p benchmarks;
           jq -s '.[0].benchmarks = ([.[].benchmarks] | add)
           | if .[0].benchmarks == null then null else .[0] end'
-          **/**/tests/*${{ matrix.package }}*-benchmark.json > benchmarks/output_${{ matrix.package }}.json
-      - name: Upload all benchmarks under same key as an artifact
+          **/**/tests/*${{ matrix.package }}*-benchmark.json > output_${{ matrix.python-version }}_${{ matrix.package }}.json
+      - name: Upload benchmarks using (${{ matrix.python-version }}, ${{ matrix.package }}) key
         if: ${{ success() }}
         uses: actions/upload-artifact@v2
         with:
-          name: benchmarks
-          path: benchmarks/output_${{ matrix.package }}.json
+          name: benchmarks_${{ matrix.python-version }}_${{ matrix.package }}
+          path: output_${{ matrix.python-version }}_${{ matrix.package }}.json
   combine-benchmarks:
     runs-on: ubuntu-latest
     needs: build
     if: ${{ always() }}
     name: Combine benchmarks from previous build job
+    strategy:
+      max-parallel: 1
```

Comment:
We run serially because the `github-action-benchmark` GH action takes a lock on the `gh-pages` branch.